### PR TITLE
Gradient.ai - updating the SDK

### PR DESCRIPTION
**Pull request by michaelfeil**
Our SDK `gradientai >= 1.4.0` supports now `async` API and threaded batching in our API - the embedding model now exposes `.embed` => This PR aims to improve speed of embeddings over large numbers of documents.

## Dependencies:
- bumped  `gradientai` from `1.3.0` to `1.4.0`.  The SDK always had a minium python version of 3.8, except for some initial experiments (e.g. `1.0.0.dev1`). Therefore, I raised the miniumum version to `1.4.0` 
- added `tqdm` as optional. It should be pre-installed, and we are using no advanced options other than `from tqdm import tqdm`, no version pin required. I think this will improve developer experience, especially since the embeddings are now faster on the server side.

@mhaligowski Can you give this a check before I mark as ready?

### CODE REVIEW

File: integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py [hunk 1]
```diff
@@ -31,15 +36,17 @@ def __init__(
         self,
         *,
         model_name: str = "bge-large",
-        batch_size: int = 100,
+        batch_size: int = 32768,
         access_token: Optional[str] = None,
         workspace_id: Optional[str] = None,
         host: Optional[str] = None,
+        progress_bar: bool = True,
```
Comment (mhaligowski) [inline integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py:43]: I think the convention is to name the variables with a "booleanish" prefix? `show_progress_bar`? `has_progress_bar`? `progress_bar` looks like it's expecting an actual object.

File: integrations/gradient/pyproject.toml [hunk 2]
```diff
@@ -7,7 +7,7 @@ name = "gradient-haystack"
 dynamic = ["version"]
 description = ''
 readme = "README.md"
-requires-python = ">=3.7"
+requires-python = ">=3.8"
```
Comment (mhaligowski) [inline integrations/gradient/pyproject.toml:10]: what's the reason for bumping?

**Comment by mhaligowski** [comment]
LGTM, let's make sure that the tests are ran and we're gtg!

File: integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py [hunk 3]
```diff
@@ -31,15 +36,17 @@ def __init__(
         self,
         *,
         model_name: str = "bge-large",
-        batch_size: int = 100,
+        batch_size: int = 32768,
```
Comment (mhaligowski) [inline integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py:39]: do you want `32_768`?

File: integrations/gradient/pyproject.toml (hunk 2 above)
Comment (michaelfeil) [reply integrations/gradient/pyproject.toml:10]: With `pip install gradientai` using a py37 environment, it installed `1.0.0.dev-something`

**Comment by michaelfeil** [comment]
All tests passing on my end.

**Comment by masci** [comment]
Thanks folks! FYI, in #103 we removed all the `__about__.py` files, to resolve the conflict just delete the one in your branch and it should be good.

**Comment by michaelfeil** [comment]
@masci Alright.

File: integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py [hunk 4]
```diff
@@ -4,9 +4,19 @@
 from gradientai import Gradient
 from haystack import Document, component, default_to_dict
 
+try:
+    from tqdm import tqdm  # type: ignore # noqa
```
Comment (michaelfeil) [inline integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py:8]: @masci Requiring some help here - the CI is failing on this line - not sure why.

File: integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py [hunk 5]
```diff
@@ -75,11 +90,17 @@ def _generate_embeddings(self, documents: List[Document], batch_size: int) -> Li
...
         Batches the documents and generates the embeddings.
         """
-        batches = [documents[i : i + batch_size] for i in range(0, len(documents), batch_size)]
+        if self._progress_bar and tqdm_imported:
+            batches = [documents[i : i + batch_size] for i in range(0, len(documents), batch_size)]
+            progress_bar = tqdm
+        else:
+            # no progress bar
+            progress_bar = _alt_progress_bar  # type: ignore
+            batches = [documents]
```
Comment (silvanocerza) [inline integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py:99]: Is this correct? It seems strange to me that batching is ignored if `tqdm` is not shown. 🤔

Comment (michaelfeil) [reply integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py:99]: Yes - our server side batch size is currently 256. 
We run a internal mapping / mirco-bacthing loop of size 256, but have no progress bar. Therefore, there is no upper limit, you could also set batch_size to 100_000

Comment (silvanocerza) [reply integrations/gradient/src/gradient_haystack/embedders/gradient_document_embedder.py:99]: Ok, good for me if it's good for you. 👍

**Review by silvanocerza — APPROVED** [review]
Cool, merging!

**Comment by michaelfeil** [comment]
Awesome, thanks guys!

**Pull request merged**